plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -805,7 +807,22 @@ public static String fileName(String path)
     public static void commit(SnapshotUpdate<?> update, ConnectorSession session)
     {
         update.set(TRINO_QUERY_ID_NAME, session.getQueryId());
-        update.commit();
+        try {
+            update.commit();
+        }
+        catch (CommitFailedException commitFailedException) {
+            throw new TrinoException(TRANSACTION_CONFLICT, commitFailedException.getMessage(), commitFailedException.getCause());
+        }
+    }
+
+    public static void commitTransaction(Transaction transaction)
+    {
+        try {
+            transaction.commitTransaction();
+        }
+        catch (CommitFailedException commitFailedException) {
+            throw new TrinoException(TRANSACTION_CONFLICT, commitFailedException.getMessage(), commitFailedException.getCause());
```

Comment:
how do we know from CommitFailedException that this is a TRANSACTION_CONFLICT situation?

what if we get CommitFailedException because eg metastore is unreachable, read-only or anything like this?